lib/amazonka/src/Amazonka/Auth/SSO.hs

```diff
@@ -0,0 +1,127 @@
+-- |
+-- Module      : Amazonka.Auth.SSO
+-- Copyright   : (c) 2013-2021 Brendan Hay
+-- License     : Mozilla Public License, v. 2.0.
+-- Maintainer  : Brendan Hay <brendan.g.hay+amazonka@gmail.com>
+-- Stability   : provisional
+-- Portability : non-portable (GHC extensions)
+module Amazonka.Auth.SSO where
+
+import Amazonka.Auth.Background (fetchAuthInBackground)
+import Amazonka.Auth.Exception
+import qualified Amazonka.Crypto as Crypto
+import Amazonka.Data.Sensitive
+import Amazonka.Data.Time (Time (..))
+import Amazonka.Env (Env, Env' (..))
+import Amazonka.HTTP (retryRequest)
+import Amazonka.Lens ((^.))
+import Amazonka.Prelude
+import Amazonka.SSO.GetRoleCredentials as SSO
+import qualified Amazonka.SSO.Types as SSO (RoleCredentials (..))
+import Amazonka.Types
+import qualified Control.Exception as Exception
+import Control.Exception.Lens (handling_, _IOException)
+import Control.Monad.Trans.Resource (runResourceT)
+import Data.Aeson (FromJSON, decodeFileStrict)
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import Data.Time.Clock.POSIX (posixSecondsToUTCTime)
+import qualified Network.HTTP.Client as Client
+
+data CachedAccessToken = CachedAccessToken
+  { startUrl :: Text,
+    region :: Region,
+    accessToken :: Sensitive Text,
+    expiresAt :: UTCTime
+  }
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (FromJSON)
+
+-- | Assume a role using an SSO Token
+--
+-- The user must have called @aws sso login@ and have appropriate @sso_@ values
+-- in their active profile, which are the arguments to this function. This
```

Comment:
This haddock is actually describing the behaviour of `fromConfigFile`. Would you mind rewriting to say something like:

```
The user must have previously called @aws sso login@, and pass in the path to the cached token file, along with SSO region, account ID and role name. ('Amazonka.Auth.ConfigFile.fromFilePath' understands the @sso_@ variables used by the official AWS CLI and will call 'fromSSO' for you.) This function uses 'fetchAuthInBackground'...
```